qemu/tests/chardev_tls_encryption.py

```diff
@@ -0,0 +1,94 @@
+        c. Check the server endpoint output
+    Scenario 2:
+        a. Launch QEMU with a serial port as TLS server
+        b. Run gnutls client to connect TLS server
+        c. Check the client endpoint output
+    Scenario 3:
+        a. Launch QEMU with a serial port as TLS server
+        b. Execute 'cat /dev/ttyS0' in guest which boot from step 1
+        c. Launch QEMU with a serial port as TLS client
+        d. Check the output of step b
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    clean_cmd = params["clean_cmd"]
+    try:
+        setup_certs(params)
+        hostname = process.run('hostname', ignore_status=False, shell=True,
+                               verbose=True).stdout_text.strip()
+        port = str(utils_misc.find_free_ports(5000, 9999, 1, hostname)[0])
+
+        # Scenario 1
+        gnutls_cmd_server = params.get("gnutls_cmd_server")
+        if gnutls_cmd_server:
+            gnutls_cmd_server = gnutls_cmd_server % port
+            params["extra_params"] = params["extra_params"] % (hostname, port)
+            tls_server = aexpect.run_bg(gnutls_cmd_server)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            env_process.preprocess_vm(test, params, env, vm_name)
+            if not utils_misc.wait_for(
+                    lambda: "Channel binding 'tls-unique'" in
+                            tls_server.get_output(), first=5, timeout=15):
+                test.fail("tls server can't connect client succssfully.")
+
+        # Scenario 2
+        gnutls_cmd_client = params.get("gnutls_cmd_client")
+        if gnutls_cmd_client:
+            gnutls_cmd_client = gnutls_cmd_client % (port, hostname)
+            params["extra_params"] = params["extra_params"] % (hostname, port)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            env_process.preprocess_vm(test, params, env, vm_name)
+            tls_client = aexpect.run_bg(gnutls_cmd_client)
+            if not utils_misc.wait_for(lambda: "Simple Client Mode:" in
+                    tls_client.get_output(), first=5, timeout=15):
+                test.fail("tls client can't connect server succssfully.")
+
+        # Scenario 3:
+        guest_cmd = params.get("guest_cmd")
+        if guest_cmd:
+            params["start_vm"] = "yes"
+            vms = params.get("vms").split()
+            params["extra_params"] = params["extra_params_vm1"] % (hostname,
+                                                                   port)
+            env_process.preprocess_vm(test, params, env, vms[0])
+            vm1 = env.get_vm(vms[0])
+            session_vm1 = vm1.wait_for_login()
+            session_vm1.cmd(guest_cmd)
+            params["extra_params"] = params["extra_params_vm2"] % (hostname,
+                                                                   port)
```

Comment:
Server and client use the same host port in the single host ?

Reply:
> Server and client use the same host port in the single host ?

yes.

Reply:
ok, thanks!